src/js/forms.js

```diff
@@ -1,201 +1,99 @@
-			this.opts.applyValidState = declaredApplyValidState === 'true';
-		}
-
-		// o-forms should only be registered against a <form>
-		// element. If not, return to prevent errors
-		if (!(this.FormEl instanceof HTMLFormElement)) {
-			this.destroy();
-			return;
-		}
-
-		// Add the event listeners
-		this.listeners();
-	}
-
-	listeners() {
-		// Setup checkbox toggle functionality.
-		// We listen for clicks as IE does not fire onchange until onblur.
-		this.FormEl.addEventListener('click', this.handleClickEvent.bind(this), false);
-
-		// Setup validity checks.
-		if (this.opts.testEvent === 'submit') {
-			// Safari reports the validity state, but doesn't
-			// prevent form submits, so this listens to submits and
-			// checks the inputs are valid before submission.
-			this.FormEl.addEventListener('submit', this.validateForm.bind(this), false);
-
-			// All other browsers will report each item invalid on
-			// submit and prevent a form submission.
-			this.findInputs().forEach(input => {
-				input.addEventListener('invalid', this.handleInvalidEvent.bind(this), false);
-			});
+		if (!this.opts.useBrowserValidation) {
+			this.formEl.setAttribute('novalidate', true);
+			this.formEl.addEventListener('submit', this);
 		} else {
-			this.findInputs().forEach((input) => {
-				input.addEventListener('blur', this.handleBlurEvent.bind(this), false);
-			});
+			this.formEl.removeAttribute('novalidate');
+			let submit = this.formEl.querySelector('input[type=submit]');
+			submit.addEventListener('click', this);
+			submit.addEventListener('keydown', this);
 		}
-	}
-
-	validateForm(event) {
-		event.preventDefault();
 
-		const checkedInputs = this.findInputs().map(input => this.validateInput(input));
-
-		if (checkedInputs.some((val) => val === false)) {
-			return;
-		}
-
-		// Complete the form submission
-		event.target.submit();
+		this.inputFields.forEach(input => {
+			input.addEventListener('blur', this);
+			input.addEventListener('input', this);
+		});
 	}
 
-	validateInput(input) {
-		if (input.checkValidity() === false) {
-			this.invalidInput(input);
-			return false;
+	/**
+	 * Event Handler
+	 * @param {Object} event - The event emitted by element/window interactions
+	 */
+	handleEvent(e) {
+		if (e.type === 'blur') {
+			this.validate(e.target);
 		}
 
-		const oFormsEl = input.closest('.o-forms');
-		const oFormsElExists = (oFormsEl instanceof HTMLElement);
-
-		if (oFormsElExists && oFormsEl.classList.contains('o-forms--error')) {
-			oFormsEl.classList.remove('o-forms--error');
+		if (e.type === 'input') {
+			let field = e.target.closest('.o-forms-input');
+			if (e.target.validity.valid && field.classList.contains(this.class.invalid)) {
+				field.classList.replace(this.class.invalid, this.class.valid);
+			}
 		}
 
-		if (this.opts && this.opts.applyValidState && oFormsElExists) {
-			oFormsEl.classList.add('o-forms--valid');
+		if (e.type === 'click' || (e.type === 'keydown' && e.key === 13)) {
```

Comment:
Minor thing: could we use a constant for this instead of a magic number? E.g. define the `KEY_RETURN` variable, then `e.key === KEY_RETURN`